test/metric/dimension/instanceid_provider.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package dimension
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/environment/computetype"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"log"
+)
+
+type ECSInstanceIdDimensionProvider struct {
+	Provider
+}
+
+var _ IProvider = (*ECSInstanceIdDimensionProvider)(nil)
+
+func (p *ECSInstanceIdDimensionProvider) IsApplicable() bool {
+	if (p.env.ComputeType == computetype.ECS) {
+		return true
+	}
+	return false
```

Comment:
equivalent to `return p.env.ComputeType == computetype.ECS`

Reply:
yes

Reply:
Done